### PR TITLE
Don't use cache when serving locally

### PR DESCRIPTION
**Pull request by jwunderl**
Sets the timeout for the cache to 1 ms when serving locally (e.g. pxt serve). Currently hard to update things as some elements like the front page get cached and you have to change the port every update.

(Looks like it still serves the cached content while it is requesting the new one, so to immediately show updates you need to refresh twice when serving locally)

### CODE REVIEW

File: pxtlib/emitter/cloud.ts [hunk 1]
```diff
@@ -90,7 +90,7 @@ namespace pxt.Cloud {
     }
 
     // 1h check on markdown content
-    const MARKDOWN_EXPIRATION = 1 * 60 * 60 * 1000;
+    const MARKDOWN_EXPIRATION = Cloud.isLocalHost() ? 1 : 1 * 60 * 60 * 1000;
```
[the remark this answers is not in the document]
Comment (jwunderl) [reply pxtlib/emitter/cloud.ts]: Don't know a way to test electron off hand, but this check should handle that case then I believe?

```typescript
Cloud.isLocalHost() && !(window as any).pxtElectron
```

**Comment by pelikhan** [comment]
In electron, you also run localhost but you want caching.

File: pxtlib/emitter/cloud.ts [hunk 2]
```diff
@@ -89,8 +89,9 @@ namespace pxt.Cloud {
         })
     }
 
+    const isElectron = !!(window as any).pxtElectron || !!(window as any).ipcRenderer;
```
Comment (pelikhan) [inline pxtlib/emitter/cloud.ts]: there should be a help on for this

Comment (jwunderl) [reply pxtlib/emitter/cloud.ts]: It's in webapp/src/electron currently which is not accessible in here, should I move that around somewhere else? Would need more testing I would assume

Comment (pelikhan) [reply pxtlib/emitter/cloud.ts]: should be in browsersutil. let's minize casting window to any (and window might be missing in nodejs)

Comment (jwunderl) [reply pxtlib/emitter/cloud.ts]: Sounds good.

File: pxtlib/emitter/cloud.ts [hunk 3]
```diff
@@ -90,7 +90,7 @@ namespace pxt.Cloud {
     }
 
     // 1h check on markdown content
-    const MARKDOWN_EXPIRATION = 1 * 60 * 60 * 1000;
+    const MARKDOWN_EXPIRATION = Cloud.isLocalHost() && !pxt.BrowserUtils.isElectron() ? 1 : 1 * 60 * 60 * 1000;
```
[the remark this answers is not in the document]
Comment (jwunderl) [reply pxtlib/emitter/cloud.ts]: Would that live in browserutils too or just in cloud?

**Comment by pelikhan** [comment]
BrowserUtil, near "isLocalHost"

**Comment by jwunderl** [comment]
> BrowserUtil, near "isLocalHost"

that lives in cloud right now, move both then?

**Pull request merged**